### PR TITLE
[cleanup] Include motion simulator vehicle data

### PR DESCRIPTION
**Pull request by marijnvdwerf**

### CODE REVIEW

File: src/ride/vehicle.c [hunk 1]
```diff
@@ -3997,8 +3997,7 @@ static void vehicle_update_simulator_operating(rct_vehicle* vehicle) {
 	if (RCT2_GLOBAL(0x00F64E34, uint8) == 0)
 		return;
 
-	uint8* edi = RCT2_ADDRESS(0x009A0434, uint8);
-	uint8 al = edi[(uint16)(vehicle->current_time + 1)];
+	uint8 al = MotionSimulatorTimeToSpriteMap[(uint16)(vehicle->current_time + 1)];
```
Comment (IntelOrca) [inline src/ride/vehicle.c]: The cast isn't a good idea anymore, I think it needs to AND it by 0xFFFF instead.

Comment (marijnvdwerf) [inline src/ride/vehicle.c]: Updated to remove the cast and add an assert. Doesn't get triggered when running the ride in test mode

Comment (IntelOrca) [inline src/ride/vehicle.c]: Shouldn't it also assert that it is below the length of the array?

Comment (marijnvdwerf) [inline src/ride/vehicle.c]: Added

**Pull request merged**